### ISSUE TITLE
[Bug] When in onboarding mode, focusing on address bar should not cause a jump to homescreen

### ISSUE THREAD

**Issue by violasong** [mozilla-mobile/fenix#16353]
Related to https://github.com/mozilla-mobile/fenix/issues/16351

Currently in Nightly, when viewing the onboarding screen and tapping the address bar, the view behind the keyboard switches from onboarding to homescreen. It's then not possible to get back to onboarding. 

Instead, opening the keyboard should not navigate away from onboarding. 

Ideally, when the keyboard is up we would adapt the same behavior as #16351, where you can still interact with the onboarding screen, and scroll on it to dismiss the keyboard.

**Comment by liuche** [comment]
@violasong could you take a look at this issue #15990 as well? I'm wondering if we should allow users to go back to onboarding once they've decided to start browsing. Does clicking on the urlbar indicate an intent to complete/skip the onboarding walkthrough?

**Comment by eliserichards** [comment]
@violasong Another edge case: 
- a user fresh installs Fenix, then goes to their Gmail and clicks on a link (opens in Fenix). At this point the user has _technically_ navigated. Do we dismiss the onboarding? Or are they still going to see it?

**Comment by violasong** [comment]
@eliserichards:
> a user fresh installs Fenix, then goes to their Gmail and clicks on a link (opens in Fenix). At this point the user has technically navigated. Do we dismiss the onboarding? Or are they still going to see it?

This will be affected by the new UX conversation about changing homescreen to a new tab so that it's part of the backstack. Once that happens, I would expect there to be two tabs in this situation - the link from Gmail and the onboarding tab.

For now, since I assume this isn't possible yet, I think we need to just dismiss the onboarding in favor of showing the link from Gmail. @apbitner and @lime124 may be able to confirm.

**Comment by apbitner** [comment]
Yeah ideally we'd open the link in a new tab and leave onboarding alone. But if that's not possible, I'm not sure what other option we have besides dismissing onboarding to open the link.
